packages/editor/src/js/vue/components/delete-block-modal/delete.modal.js

```diff
@@ -0,0 +1,100 @@
+const Vue = require('vue/dist/vue.common');
+const { ModalComponent } = require('../modal/modalComponent');
+const { deletePersonalizedBlock } = require('../../utils/apis');
+const axios = require('axios');
+const styleHelper = require('../../utils/style/styleHelper');
+
+const DeleteBlockModalComponent = Vue.component('DeleteBlockModal', {
+  components: {
+    ModalComponent,
+  },
+  props: {
+    vm: { type: Object, default: () => ({}) },
+  },
+  data: () => ({
+    blockId: null,
+    blockName: '',
+    isLoading: false,
+    style: styleHelper,
+  }),
+  mounted() {
+    this.vm.toggleDeleteBlockModal = this.handleToggleDeleteBlockModalChange;
+  },
+  computed: {
+    disableDeleteButton() {
+      return this.isLoading || !this.blockId;
+    },
+  },
+  methods: {
+    openModal() {
+      this.$refs.modalRef?.openModal();
+    },
+    closeModal() {
+      this.blockId = null;
+      this.$refs.modalRef?.closeModal();
+    },
+    handleToggleDeleteBlockModalChange(value, blockContent) {
+      const blockId = blockContent?.blockInformation?.id;
+      this.blockName = blockContent?.blockInformation?.name;
+      if (value === true) {
+        this.openModal();
+        this.blockId = blockId;
+      } else {
+        this.closeModal();
+        this.blockId = null;
+      }
+    },
+    handleOnSubmit() {
+      this.isLoading = true;
+      const groupId = this.vm?.metadata?.groupId;
+      axios
+        .delete(deletePersonalizedBlock(this.blockId), { params: { groupId } })
+        .then(() => {
+          this.vm.notifier.success(this.vm.t('delete-block-success'));
+          this.isLoading = false;
+          this.closeModal();
+          const event = new Event('personalizedBlockApiActionApplied');
+          window.dispatchEvent(event);
+        })
+        .catch(() => {
+          this.vm.notifier.error(this.vm.t('delete-block-error'));
+          this.isLoading = false;
+        });
+    },
+  },
+  template: `
+  <modal-component ref="modalRef" :class="{ 'small-modal': true, 'error-modal': true }">
```

Comment:
```suggestion
  <modal-component ref="modalRef" class="small-modal error-modal">
```

Because here this isn't dynamic ;)